### ISSUE TITLE
Limitations of the kwargs_to_strings decorator not handling positional arguments

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/pygmt#2361]
Here are some limitations of the `kwargs_to_strings` decorator:

1. Doesn't work for the default values of parameters:

https://github.com/GenericMappingTools/pygmt/pull/2208#discussion_r1038274212

> Please note that we can't use decorator `@kwargs_to_strings(offset="sequence")` here, because the decorator can only deal with keyword arguments and doesn't work for the default values. I.e., it doesn't work for `Figure.timestamp()` but works for `Figure.timestamp(offset=["-54p", "-54p"])`, although offset defaults to `["-54p", "-54p"]`.

```python
@kwargs_to_strings(offset="sequence")
def timestamp(
    self,
    text=None,
    label=None,
    justification="BL",
    offset=("-54p", "-54p"),
    font="Helvetica",
    timefmt="%Y %b %d %H:%M:%S",
):
```

2. Doesn't work if the parameter name is not given
```python
>>> from pygmt import which
>>> # doesn't work
>>> which(["@hotspots.txt", "@earth_relief_01d_g"])
gmtwhich [ERROR]: File ['@hotspots.txt', not found!
gmtwhich [ERROR]: File '@earth_relief_01d_g'] not found!
>>> # works
>>> which(fname=["@hotspots.txt", "@earth_relief_01d_g"])
```

**Comment by weiji14** [comment]
This issue would also apply to `fig.text`'s `textfiles` parameter that uses `sequence_space` added in https://github.com/GenericMappingTools/pygmt/pull/325.

https://github.com/GenericMappingTools/pygmt/blob/c9d6147bcf383903ffc75fb94f9787b6b3a1d231/pygmt/src/text.py#L41-L50

Should we remove `sequence_space` entirely since it is only used in `pygmt.which` and `fig.text`? And use the `if is_nonstr_iter(fname): fname = " ".join(fname)` patch in #2726?

**Comment by seisman** [comment]
> And use the `if is_nonstr_iter(fname): fname = " ".join(fname)` patch in #2726?

No, it won't work for `Figure.text`, because `textfiles` can be a 2-D array, e.g., `textfiles=[[1, 2, "A"], [3, 4, "B"]]`.

**Comment by weiji14** [comment]
> > And use the `if is_nonstr_iter(fname): fname = " ".join(fname)` patch in #2726?
> 
> No, it won't work for `Figure.text`, because `textfiles` can be a 2-D array, e.g., `textfiles=[[1, 2, "A"], [3, 4, "B"]]`.

Ah ok, you're right. Let's keep `sequence_space` then. So #2726 will fix this issue entirely? Can we change `Addresses #2726` to `Fixes #2726` or was there something else?

**Comment by seisman** [comment]
> So #2726 will fix this issue entirely?

No, see the first point:

> Doesn't work for the default values of parameters:
